### ISSUE TITLE
Move Damage/Accuracy randomization

### ISSUE THREAD

**Issue by Bobbsify** [kwsch/pk3DS#414]
Greetings Fellow Humans! 👋
I was reading through issue #287 [Regarding damage randomization on moves]
I understand the problems @kwsch has addressed such as:
>How do status moves get randomized?
>
and
>What extremes should values be randomized to?
>
and was wondering why wouldn't it just be handled like [Universal Pokèmon Randomizer](https://pokehacks.dabomstew.com/randomizer/) (**UPR**) does for the previous gens:

- Status moves will only get randomized accuracy, no other effect as they do not have an original damage stat.

- The extremes to which a move should be randomized have been laid out by the original developers of the game with 10 Being the lowest damage and 250 being the highest
  - Although I have seen from moves from **UPR** that came out with damage going down to 5, so honestly i don't really know how it's handled there.

- IF you are willing to randomize damage you should be prepared to face moves of nonsensical power such as a 250 power Stormthrow

- You shouldn't be restricted to have low power moves starting out, that's the fun of it!
  - You could get blasted in the face by a 95 Accuracy Sheercold  or you could blast people away with a 180 Power Scald!
  - My point being that huge highs and bigger lows are the fun of a Randomized run, that's why you randomize in the first place! You shouldn't limit low level pokèmon to only have low power moves as that negates some extremely fun moments.

I have been also trying to write a function that tries to randomize a stat but keeping it somewhat close to the starting one, it looks something like this:
```C#
public int getRandomMoveStat(int originalStat){
            int decrChance = -5;
            int newStat = originalStat; 

            do
            {
                newStat = new Random().nextInt(6)%2 == 0 ? newStat+5:newStat-5;
                decrChance+=5;                
            }while(new Random().nextInt(101)<100-decrChance);

            return newStat;
        }

```

## TL;DR: Why Isn't damage and accuracy randomization handled like Universal Pokèmon Randomizer? ##

**Comment by kwsch** [comment]
>Why Isn't damage and accuracy randomization handled like UPR?

Because pk3DS only changes category and type; move power and accuracy are unchanged. Modifying more than that doesn't give much more benefit for randomization sake.

Random mutations won't work effectively with the example code you've provided. On average, the values would be unchanged; 50% chance to increase and a 50% chance to decrease, breaking eventually.

Here's the code that UPR uses to randomize the following:

* Power
* PP
* Accuracy
* Type
* Category

https://github.com/Dabomstew/universal-pokemon-randomizer/blob/c16ce0ad253c4cebbfc25ac0e566d4f13d81b283/src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java#L1090-L1215

**Comment by tommaniacal** [comment]
How is randomizing power and accuracy any different than randomizing base stats? Couldn't you do something similar to what pk3ds already does, and limit the deviance to say, 50%? OHKO moves's accuracy would never go above 45%. You could also add a cap for absurdly powerful moves like Z-moves and Explosion, capping it at 250 bp.

Doesn't the game already have a chance to have powerful early game moves? It's very possible that your rival's pokemon can start out with Draco Meteor or V Create. Randomizing the power doesn't make this any better or worse.

Having an option to randomize power and accuracy isn't any less smart than not having an option to re-order damaging moves

If someone chooses to randomize move power and accuracy, they're acknowledging the fact that it might be frustrating and end early. They can choose not to randomize them.

**Comment by kwsch** [comment]
Closing stale issue due to no progress in implementation.

Pull requests implementing the requested features will be accepted.
